src/DotVVM.Framework/ResourceManagement/LocalResourceUrlManager.cs

```diff
@@ -77,20 +116,36 @@ private ILocalResourceLocation TryLoadAlternativeFile(string name, string hash,
                 if (directory != null)
                 {
                     var sourceFile = Path.Combine(directory, name);
-                    if (File.Exists(sourceFile)) return new FileResourceLocation(sourceFile);
+                    if (File.Exists(sourceFile))
+                    {
+                        return new FileResourceLocation(sourceFile);
+                    }
                 }
             }
             return null;
         }
 
-        protected ILocalResourceLocation FindLocation(IResource resource, out string mimeType)
+        protected ILocalResourceLocation FindLocation(
+            IResource resource,
+            int? locationIndex,
+            out string mimeType)
         {
-            var linkResource = resource as ILinkResource;
-            mimeType = linkResource?.MimeType;
-            return linkResource
-                   ?.GetLocations()
-                   ?.OfType<ILocalResourceLocation>()
-                   ?.FirstOrDefault();
+            if (!(resource is ILinkResource link))
+            {
+                mimeType = null;
+                return null;
+            }
+
+            mimeType = link.MimeType;
+            var locations = link.GetLocations().ToArray();
+            if (locationIndex == null)
+            {
+                return locations
+                    .OfType<ILocalResourceLocation>()
+                    .FirstOrDefault();
+            }
+            // unless the order of locations changes, the location has to be local
```

Comment:
And when the order of locations changes then what? What does this comment says?

Reply:
It meant to say, that the type of the location is guaranteed to be `ILocalResourceLocation`, which isn't true anyway. I added a condition instead.